### ISSUE TITLE
aria-label warning with ng-switch

### ISSUE THREAD

**Issue by calebegg** [angular/material#8237]
Demo:

http://codepen.io/calebegg/pen/bpmGqQ?editors=1010

This code triggers a warning about aria-label being missing on the button, even though the button is guaranteed to have text content, since there's an ng-switch-default.
```html
    <md-button ng-switch on="'x'">
      <span ng-switch-when="x">x</span>
      <span ng-switch-default>y</span>
    </md-button>
```

**Comment by DevVersion** [comment]
That seems to be a problem, with the DOM modification. Since the switch elements are appended dynamically to the DOM, we're not updating our label text.

I guess, using a Observer would be overkill.

**Comment by ThomasBurleson** [comment]
Edge case - will not fix for ngMaterial v1.x.
